ProcessQueue/__init__.py

```diff
@@ -32,27 +32,42 @@ def process_scan_report_sheet_table(sheet):
     # Get max number of columns in the sheet
     max_column=sheet.max_column
     # Set a column index (openpyxl index at 1 instead of 0)
-    col_idx=1
     
     for row_idx,row_cell in enumerate(sheet.iter_rows(min_row=1,max_col=max_column),start = 1):
   
     # Works through pairs of value/frequency columns
         for column_idx,cell in enumerate(row_cell):

```

Comment:
What is the pseudocode for this process? I feel like it could be tidied up somewhat. Is it this?
```
Loop over all columns. Skip the frequency columns, only process the value columns.
  For each value column, loop over all rows - if either the value column or its partnered frequency column is non-empty, then process this.
    Append the (column-name,value,frequency) tuple to results 
```

Reply:
In fact, I think we should be looping over columns first, and then rows (because columns have different lengths).
```
for col_idx, column in enumerate(sheet.iter_cols(min_row=1, max_row=sheet.max_row), start=1):
  if col_idx % 2  == 0:
    continue
  for row_idx, cell in enumerate(column):
    if cell.value == "" and sheet.cell(row=row_idx,column=column_idx+2).value == "":
      continue
    results.append(  (sheet.cell(row=1,column=column_idx+1).value, cell.value, sheet.cell(row=row_idx,column=column_idx+2).value)
```

Reply:
(I have not tested the above)

Reply:
```
for every row in a sheet:
  for every column in the row:
      If cell.value is not empty then:
         make sure to take pairs of value/frequency columns (%2=0)
          if both value and frequency are empty, continue (this can be removed I think, if cell.value takes care of this)
           append the (column-name,value,frequency) to results 
      Else, 
             if cell.value is empty but there is a frequency
               append column-name,value==empty,frequency) to results 

```

Reply:
I think we should certainly move the `(%2=0)` check as high as possible because we want to ignore the frequency columns immediately, and just process the value columns (with the associated freq of course). And combine the check for a value in the cell with the check for the value in the associated frequency - we only do something if either is non-empty, otherwise we ignore.

(What happens if value is nonempty but frequency is empty, or can that never happen?)

Reply:
if there is a value but no frequency I set it to be 0 in line 273-276

Reply:
if we want it to be empty instead of zero we can allow it to be blank like we have the field, value names

Reply:
Ok, I'm happy to set it to zero if it's empty. So if either the value or the freq is non-empty then we should add this value-freq pair to `results`, and we only `continue` if both are empty - is that correct?

I think it's conceptually clearer to write it as 
```
# Iterate over all columns in this sheet
for col_idx, column in enumerate(sheet.iter_cols(min_row=1, max_row=sheet.max_row), start=1):
  # Only process the value columns, and skip the frequency columns
  if col_idx % 2  == 0:
    continue
  # Loop over all rows in this column
  for row_idx, cell in enumerate(column):
    # Skip this value if it and its frequency are empty
    if cell.value == "" and sheet.cell(row=row_idx,column=column_idx+2).value == "":
      continue
    # Otherwise, one of the value-freq pair must be non-empty, and so we add the value-freq pair to results
    results.append((sheet.cell(row=1,column=column_idx+1).value, cell.value, sheet.cell(row=row_idx,column=column_idx+2).value))
```